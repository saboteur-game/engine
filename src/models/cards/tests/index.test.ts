import { getShuffledDeck } from "..";

jest.spyOn(global.Math, "random").mockReturnValue(0.5);

describe("cards", () => {
  describe("getShuffledDeck", () => {
    it("returns a full deck", () => {
      const deck = getShuffledDeck();
      expect(deck.length).toBe(67);
      expect(deck).toMatchSnapshot();
    });
  });
});
