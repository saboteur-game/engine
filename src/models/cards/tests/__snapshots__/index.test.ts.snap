// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cards getShuffledDeck returns a full deck 1`] = `
Array [
  PassageCard {
    "connectors": Array [
      2,
      4,
    ],
    "id": "test-id-0",
    "isUpsideDown": false,
    "status": "unused",
  },
  PassageCard {
    "connectors": Array [
      2,
      4,
    ],
    "id": "test-id-1",
    "isUpsideDown": false,
    "status": "unused",
  },
  PassageCard {
    "connectors": Array [
      2,
      4,
    ],
    "id": "test-id-2",
    "isUpsideDown": false,
    "status": "unused",
  },
  PassageCard {
    "connectors": Array [
      1,
      3,
    ],
    "id": "test-id-3",
    "isUpsideDown": false,
    "status": "unused",
  },
  PassageCard {
    "connectors": Array [
      1,
      3,
    ],
    "id": "test-id-4",
    "isUpsideDown": false,
    "status": "unused",
  },
  PassageCard {
    "connectors": Array [
      1,
      3,
    ],
    "id": "test-id-5",
    "isUpsideDown": false,
    "status": "unused",
  },
  PassageCard {
    "connectors": Array [
      1,
      3,
    ],
    "id": "test-id-6",
    "isUpsideDown": false,
    "status": "unused",
  },
  PassageCard {
    "connectors": Array [
      1,
      4,
    ],
    "id": "test-id-7",
    "isUpsideDown": false,
    "status": "unused",
  },
  PassageCard {
    "connectors": Array [
      1,
      4,
    ],
    "id": "test-id-8",
    "isUpsideDown": false,
    "status": "unused",
  },
  PassageCard {
    "connectors": Array [
      1,
      4,
    ],
    "id": "test-id-9",
    "isUpsideDown": false,
    "status": "unused",
  },
  PassageCard {
    "connectors": Array [
      1,
      4,
    ],
    "id": "test-id-10",
    "isUpsideDown": false,
    "status": "unused",
  },
  PassageCard {
    "connectors": Array [
      1,
      2,
    ],
    "id": "test-id-11",
    "isUpsideDown": false,
    "status": "unused",
  },
  PassageCard {
    "connectors": Array [
      1,
      2,
    ],
    "id": "test-id-12",
    "isUpsideDown": false,
    "status": "unused",
  },
  PassageCard {
    "connectors": Array [
      1,
      2,
    ],
    "id": "test-id-13",
    "isUpsideDown": false,
    "status": "unused",
  },
  PassageCard {
    "connectors": Array [
      1,
      2,
    ],
    "id": "test-id-14",
    "isUpsideDown": false,
    "status": "unused",
  },
  PassageCard {
    "connectors": Array [
      1,
      2,
    ],
    "id": "test-id-15",
    "isUpsideDown": false,
    "status": "unused",
  },
  PassageCard {
    "connectors": Array [
      1,
      2,
      4,
    ],
    "id": "test-id-16",
    "isUpsideDown": false,
    "status": "unused",
  },
  PassageCard {
    "connectors": Array [
      1,
      2,
      4,
    ],
    "id": "test-id-17",
    "isUpsideDown": false,
    "status": "unused",
  },
  PassageCard {
    "connectors": Array [
      1,
      2,
      4,
    ],
    "id": "test-id-18",
    "isUpsideDown": false,
    "status": "unused",
  },
  PassageCard {
    "connectors": Array [
      1,
      2,
      4,
    ],
    "id": "test-id-19",
    "isUpsideDown": false,
    "status": "unused",
  },
  PassageCard {
    "connectors": Array [
      1,
      2,
      4,
    ],
    "id": "test-id-20",
    "isUpsideDown": false,
    "status": "unused",
  },
  PassageCard {
    "connectors": Array [
      1,
      2,
      3,
    ],
    "id": "test-id-21",
    "isUpsideDown": false,
    "status": "unused",
  },
  PassageCard {
    "connectors": Array [
      1,
      2,
      3,
    ],
    "id": "test-id-22",
    "isUpsideDown": false,
    "status": "unused",
  },
  PassageCard {
    "connectors": Array [
      1,
      2,
      3,
    ],
    "id": "test-id-23",
    "isUpsideDown": false,
    "status": "unused",
  },
  PassageCard {
    "connectors": Array [
      1,
      2,
      3,
    ],
    "id": "test-id-24",
    "isUpsideDown": false,
    "status": "unused",
  },
  PassageCard {
    "connectors": Array [
      1,
      2,
      3,
    ],
    "id": "test-id-25",
    "isUpsideDown": false,
    "status": "unused",
  },
  PassageCard {
    "connectors": Array [
      1,
      2,
      3,
      4,
    ],
    "id": "test-id-26",
    "isUpsideDown": false,
    "status": "unused",
  },
  PassageCard {
    "connectors": Array [
      1,
      2,
      3,
      4,
    ],
    "id": "test-id-27",
    "isUpsideDown": false,
    "status": "unused",
  },
  PassageCard {
    "connectors": Array [
      1,
      2,
      3,
      4,
    ],
    "id": "test-id-28",
    "isUpsideDown": false,
    "status": "unused",
  },
  PassageCard {
    "connectors": Array [
      1,
      2,
      3,
      4,
    ],
    "id": "test-id-29",
    "isUpsideDown": false,
    "status": "unused",
  },
  PassageCard {
    "connectors": Array [
      1,
      2,
      3,
      4,
    ],
    "id": "test-id-30",
    "isUpsideDown": false,
    "status": "unused",
  },
  DeadendCard {
    "connectors": Array [
      1,
    ],
    "id": "test-id-31",
    "isUpsideDown": false,
    "status": "unused",
  },
  DeadendCard {
    "connectors": Array [
      2,
    ],
    "id": "test-id-32",
    "isUpsideDown": false,
    "status": "unused",
  },
  DeadendCard {
    "connectors": Array [
      1,
      2,
    ],
    "id": "test-id-33",
    "isUpsideDown": false,
    "status": "unused",
  },
  DeadendCard {
    "connectors": Array [
      1,
      3,
    ],
    "id": "test-id-34",
    "isUpsideDown": false,
    "status": "unused",
  },
  DeadendCard {
    "connectors": Array [
      1,
      4,
    ],
    "id": "test-id-35",
    "isUpsideDown": false,
    "status": "unused",
  },
  DeadendCard {
    "connectors": Array [
      2,
      4,
    ],
    "id": "test-id-36",
    "isUpsideDown": false,
    "status": "unused",
  },
  DeadendCard {
    "connectors": Array [
      1,
      2,
      3,
    ],
    "id": "test-id-37",
    "isUpsideDown": false,
    "status": "unused",
  },
  DeadendCard {
    "connectors": Array [
      1,
      2,
      4,
    ],
    "id": "test-id-38",
    "isUpsideDown": false,
    "status": "unused",
  },
  DeadendCard {
    "connectors": Array [
      1,
      2,
      3,
      4,
    ],
    "id": "test-id-39",
    "isUpsideDown": false,
    "status": "unused",
  },
  MapActionCard {
    "id": "test-id-40",
    "status": "unused",
  },
  MapActionCard {
    "id": "test-id-41",
    "status": "unused",
  },
  MapActionCard {
    "id": "test-id-42",
    "status": "unused",
  },
  MapActionCard {
    "id": "test-id-43",
    "status": "unused",
  },
  MapActionCard {
    "id": "test-id-44",
    "status": "unused",
  },
  MapActionCard {
    "id": "test-id-45",
    "status": "unused",
  },
  RockfallActionCard {
    "id": "test-id-46",
    "status": "unused",
  },
  RockfallActionCard {
    "id": "test-id-47",
    "status": "unused",
  },
  RockfallActionCard {
    "id": "test-id-48",
    "status": "unused",
  },
  BreakActionCard {
    "appliesTo": Array [
      "pick",
    ],
    "id": "test-id-49",
    "status": "unused",
  },
  BreakActionCard {
    "appliesTo": Array [
      "pick",
    ],
    "id": "test-id-50",
    "status": "unused",
  },
  BreakActionCard {
    "appliesTo": Array [
      "pick",
    ],
    "id": "test-id-51",
    "status": "unused",
  },
  BreakActionCard {
    "appliesTo": Array [
      "lamp",
    ],
    "id": "test-id-52",
    "status": "unused",
  },
  BreakActionCard {
    "appliesTo": Array [
      "lamp",
    ],
    "id": "test-id-53",
    "status": "unused",
  },
  BreakActionCard {
    "appliesTo": Array [
      "lamp",
    ],
    "id": "test-id-54",
    "status": "unused",
  },
  BreakActionCard {
    "appliesTo": Array [
      "wagon",
    ],
    "id": "test-id-55",
    "status": "unused",
  },
  BreakActionCard {
    "appliesTo": Array [
      "wagon",
    ],
    "id": "test-id-56",
    "status": "unused",
  },
  BreakActionCard {
    "appliesTo": Array [
      "wagon",
    ],
    "id": "test-id-57",
    "status": "unused",
  },
  RepairActionCard {
    "appliesTo": Array [
      "pick",
    ],
    "id": "test-id-58",
    "status": "unused",
  },
  RepairActionCard {
    "appliesTo": Array [
      "pick",
    ],
    "id": "test-id-59",
    "status": "unused",
  },
  RepairActionCard {
    "appliesTo": Array [
      "lamp",
    ],
    "id": "test-id-60",
    "status": "unused",
  },
  RepairActionCard {
    "appliesTo": Array [
      "lamp",
    ],
    "id": "test-id-61",
    "status": "unused",
  },
  RepairActionCard {
    "appliesTo": Array [
      "wagon",
    ],
    "id": "test-id-62",
    "status": "unused",
  },
  RepairActionCard {
    "appliesTo": Array [
      "wagon",
    ],
    "id": "test-id-63",
    "status": "unused",
  },
  RepairActionCard {
    "appliesTo": Array [
      "pick",
      "lamp",
    ],
    "id": "test-id-64",
    "status": "unused",
  },
  RepairActionCard {
    "appliesTo": Array [
      "pick",
      "wagon",
    ],
    "id": "test-id-65",
    "status": "unused",
  },
  RepairActionCard {
    "appliesTo": Array [
      "lamp",
      "wagon",
    ],
    "id": "test-id-66",
    "status": "unused",
  },
]
`;
